decoder/vaapidecoder_vc1.cpp

```diff
@@ -277,8 +278,13 @@ bool VaapiDecoderVC1::ensurePicture(PicturePtr& picture)
     }
 
     if (frameHdr->picture_type == FRAME_P
-        || frameHdr->picture_type == FRAME_SKIPPED)
-        param->forward_reference_picture = m_forwardPicture->getSurfaceID();
+        || frameHdr->picture_type == FRAME_SKIPPED) {
+        param->forward_reference_picture = m_dpb[m_dpbIdx-1]->getSurfaceID();
```

Comment:
This will crash if the first frame is P frame, do we have some check to make sure first frame is not P frmae